homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -116,6 +150,63 @@ async def async_step_user(
             data_schema=vol.Schema(
                 {
                     vol.Required(CONF_ADDRESS): str,
+                    vol.Required(CONF_PIN): str,
+                },
+            ),
+        )
+
+    async def async_step_reauth(
+        self, entry_data: Mapping[str, Any]
+    ) -> ConfigFlowResult:
+        """Perform reauthentication upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Confirm reauthentication dialog."""
+        errors: dict[str, str] = {}
+
+        if user_input:
+            channel_id = random.randint(1, 0xFFFFFFFF)
+            self.address = user_input[CONF_ADDRESS]
+            self.pin = user_input[CONF_PIN]
+
+            pin = int(self.pin) if self.pin else None
+            mower = Mower(channel_id, self.address, pin)
+
+            try:
+                device = bluetooth.async_ble_device_from_address(
+                    self.hass, self.address, connectable=True
+                ) or await get_device(self.address)
+
+                if not await mower.connect(device):
+                    errors["base"] = "invalid_auth"
+
+                    return self.async_show_form(
+                        step_id="reauth_confirm",
+                        data_schema=vol.Schema(
+                            {
+                                vol.Required(CONF_ADDRESS): str,
+                                vol.Required(CONF_PIN): str,
+                            },
+                        ),
+                        errors=errors,
+                    )
+            except (TimeoutError, BleakError):
+                return self.async_abort(reason="cannot_connect")
+
+            await self.async_set_unique_id(self.address, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
```

Comment:
if it's reauth, we should abort if the existing unique id mismatches with the one configured